### ISSUE TITLE
Error in alevin-fry quant

### ISSUE THREAD

**Issue by alexdobin** [COMBINE-lab/alevin-fry#18]
Hi alevin-fry Developers,

I am getting the following error when running ```alevin-fry quant```:
```
thread 'main' panicked at 'could not quantify rad file.: Error(Deserialize { pos: Some(Position { byte: 0, line: 1, record: 0 }), err: DeserializeError { field: None, kind: Message("invalid length 1, expected a tuple of size 2") } })', src/main.rs:286:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
salmon 1.4.0
alevin-fry 0.1.0 from bioconda

I am following the alevin-fry tutorial with the following commands:
```
salmon alevin -l ISR --sketch --rad --chromiumV3 -p 20 -o ./map -i ./index/ --tgMap ./transcript_to_gene.2col.txt -1 ./pbmc_10k_v3_S1_L001_R1_001.fastq.gz -2 ./pbmc_10k_v3_S1_L001_R2_001.fastq.gz
alevin-fry generate-permit-list -d fw -k -i ./map -o ./quant
alevin-fry collate -t 20 -i ./quant -r ./map
alevin-fry quant   -t 20 -i ./quant -o ./quant --resolution cr-like --use-mtx --tg-map ./transcript_to_gene.2col.txt
```
The first 3 commands complete successfully. Full stdout is attached:
[log.txt](https://github.com/COMBINE-lab/alevin-fry/files/6046022/log.txt)

Thanks!
Alex

**Comment by rob-p** [comment]
Hi @alexdobin,

  Thanks for the detailed report.  We’d be happy to look into this.  The error seems especially strange since (judging by the read names) we’ve run this sample many times before.  Do you think it might be possible to share your collated RAD file (i.e. the contents of the `./map` directory right before you run the `quant` command) and the file `transcript_to_gene.2col.txt`?  This may help us debug further & more quickly.  Also, you may get a better backtrack if you run with `RUST_BACKTRACE=full` (though it still won’t be too pretty because the binary is built without debug symbols).

Thanks!
Rob

**Comment by alexdobin** [comment]
Hi Rob,

thanks a lot for the quick reply!
Here are the files you requested: http://labshare.cshl.edu/shares/dobin/dobin/alevin-fry_issue2/
```log_quant_BTfull``` contains the backtrace of the ```alevin-fry quant```.
Happy to run more tests!

Cheers,
Alex

**Comment by rob-p** [comment]
Hi @alexdobin,

  Thanks so much!  So I ran with your data and was immediately able to reproduce your original error / issue.  Squinting at the error a bit, there was an issue in "deserialization" where two tokens were expected but one was observed.  We use deserialization in 2 places in `quant`, for the RAD file, and for the transcript-to-gene map.  

  As it turns out your collated RAD file was fine; the issue is that the transcript-to-gene mapping was space separated, but needs to be tab separated.  I converted the file and then quantification ran without issue.  We should *absolutely* produce a more meaningful error message in this case, as it was not at all obvious this was the problem.  Either way, I'm glad the resolution turned out to be simple here.  Let me know if this fixes the issue on your end.

Best,
Rob
